Index: src/java/org/datanucleus/metadata/xml/PersistenceFileMetaDataHandler.java
===================================================================
--- src/java/org/datanucleus/metadata/xml/PersistenceFileMetaDataHandler.java	(revision 6299)
+++ src/java/org/datanucleus/metadata/xml/PersistenceFileMetaDataHandler.java	Tue Jul 07 16:58:09 PDT 2009
@@ -55,33 +55,29 @@
         metadata = new PersistenceFileMetaData(filename);
         pushStack(metadata);
 
+        String rootFilename;
         // Strip off "META-INF/persistence.xml" to get the root URL
         if (filename.endsWith("/META-INF/persistence.xml")) // Presumably the same on Windows?
         {
-            String rootFilename = filename.substring(0, filename.length() - "/META-INF/persistence.xml".length());
-            try
-            {
-                rootURI = new URI(rootFilename);
+            rootFilename = filename.substring(0, filename.length() - "/META-INF/persistence.xml".length());
-            }
+        }
-            catch (URISyntaxException e)
-            {
-                NucleusLogger.METADATA.warn("Error deriving persistence-unit root URI from " + rootFilename, e);
-            }
-        }
         else
         {
             // User-defined "persistence.xml" location so go up one level (omit "/persistence-filename")
-            String rootFilename = filename.substring(0, filename.lastIndexOf("/"));
+            rootFilename = filename.substring(0, filename.lastIndexOf("/"));
+        }
+
+        rootFilename = rootFilename.replace(" ", "%20");
+
-            try
-            {
-                rootURI = new URI(rootFilename);
-            }
-            catch (URISyntaxException e)
-            {
-                NucleusLogger.METADATA.warn("Error deriving persistence-unit root URI from " + rootFilename, e);
-            }
-        }
+        try
+        {
+            rootURI = new URI(rootFilename);
+        }
+        catch (URISyntaxException e)
+        {
+            NucleusLogger.METADATA.warn("Error deriving persistence-unit root URI from " + rootFilename, e);
+        }
+    }
-    }
 
     /**
      * Handler method called at the start of an element.
